src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -187,7 +187,8 @@ public String greeting(
         if (!bookToUpdate.getId().equals(id)) {
             throw new BookIdMismatchException(MSGIDMISMATCH);
         }
-        if (bookRepository.findBookByAuthor(bookToUpdate.getAuthor()).isPresent()){
+        if ((bookRepository.findBookByAuthor(bookToUpdate.getAuthor()).isPresent())&&
+                (bookRepository.findBookByAuthor(bookToUpdate.getAuthor())).get().getId() != id){
```

Comment:
you can check directly by finding the book by id :)

Reply:
I do not know how apply the suggest :cry: